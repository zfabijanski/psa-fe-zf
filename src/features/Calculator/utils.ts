import every from "lodash/every";
import isEmpty from "lodash/isEmpty";
import isObject from "lodash/isObject";
import mergeWith from "lodash/mergeWith";
import { ProductCategory } from "../../models/common";
import { CalculationIdType } from "../Products/types";
import { defaultValues } from "./defaultValues";
import { ICalculationVM, ICover, IInsured } from "./types";
import {
  CalculationState,
  CalculationStatus,
  Direction,
} from "models/calculator";
import { Frequency, NonStandardFrequency } from "models/common";
import { FrequencyToMultiplier } from "./validationSchema/coverSchema";
import { formatNumber } from "utils/transformers";

export const mergeWithDefaultValues = (
  values: Partial<ICalculationVM> & {
    productGuid: number;
    productCategories: ProductCategory[];
  }
): ICalculationVM =>
  mergeWith({}, defaultValues, values, (defaultValue, apiValue) => {
    if (Array.isArray(defaultValue) && every(defaultValue, isObject)) {
      return defaultValue.map((currentValue, index) =>
        mergeWith({}, (apiValue || [])[index], currentValue, (x, y) => x || y)
      );
    }

    return apiValue || defaultValue;
  });

export function isCalculation(
  calculation: ICalculationVM | CalculationIdType
): boolean {
  return isObject(calculation) && !isEmpty(calculation);
}

export function getCalculationId(calculation: number | ICalculationVM): number;
export function getCalculationId(
  calculation?: CalculationIdType | ICalculationVM
): number | undefined;
export function getCalculationId(
  calculation: CalculationIdType | ICalculationVM | undefined
) {
  if (!calculation) {
    return;
  }

  if (typeof calculation === "number") {
    return calculation;
  }

  return calculation.calculationId;
}

export function getCalculationStatus(
  calculationsState: { [calculationId: number]: CalculationState },
  calculation?: number | CalculationIdType | ICalculationVM
) {
  const calculationId = getCalculationId(calculation);
  if (!calculationId || !calculationsState[calculationId]) {
    return CalculationStatus.NotCalculated;
  }
  return (
    calculationsState[calculationId].status || CalculationStatus.NotCalculated
  );
}

const getCoverCopy = (cover: ICover, isMain?: boolean): ICover => ({
  ...cover,
  sumAssured:
    cover.directionOfCalculation === Direction.SumAssuredToPremium
      ? cover.sumAssured
      : undefined,
  addSumAssured:
    isMain || cover.directionOfCalculation === Direction.SumAssuredToPremium
      ? cover.addSumAssured
      : undefined,
  premiumPerFreq:
    cover.directionOfCalculation === Direction.PremiumToSumAssured
      ? cover.premiumPerFreq
      : undefined,
  expMatBen:
    cover.directionOfCalculation === Direction.MaturityBenefitToPremium
      ? cover.expMatBen
      : undefined,
});

const getInsuredCopy = (insured: IInsured): IInsured => ({
  ...insured,
  covers: insured.covers && insured.covers.map((cover) => getCoverCopy(cover)),
});

export const getCalculationCopy = (
  calculation: ICalculationVM
): ICalculationVM => ({
  ...calculation,
  calculationId: undefined,
  premiumsTotal: undefined,
  projectedMaturityValueEgrM: undefined,
  guaranteedMaturityBenefit: undefined,
  guaranteedDeathBenefit: undefined,
  mainInsured: getInsuredCopy(calculation.mainInsured),
  policyHolder:
    calculation.policyHolder && getInsuredCopy(calculation.policyHolder),
  additionalLifeAssureds:
    calculation.additionalLifeAssureds &&
    calculation.additionalLifeAssureds.map(getInsuredCopy),
  mainCover: getCoverCopy(calculation.mainCover, true),
  additionalCovers:
    calculation.additionalCovers &&
    calculation.additionalCovers.map((cover) => getCoverCopy(cover)),
  wopCovers:
    calculation.wopCovers &&
    calculation.wopCovers.map((cover) => getCoverCopy(cover)),
});

export const mapDictionaryValues = <T extends object, U>(
  dictionary: T,
  callback: (value: U) => U
) =>
  Object.fromEntries(
    Object.entries(dictionary).map(([key, value]) => [key, callback(value)])
  );

export const filterDictionaryByValue = <T extends object, U>(
  dictionary: T,
  callback: (value: U) => boolean
) =>
  Object.fromEntries(
    Object.entries(dictionary).filter(([key, value]) => callback(value))
  );

export const getResolveParameter =
  (
    name: string,
    min?: number,
    max?: number,
    frequency?: Frequency | NonStandardFrequency
  ) =>
  (paramName: string) => {
    const getMinValue = (
      val?: number,
      frequency?: Frequency | NonStandardFrequency.None
    ) =>
      name && frequency && frequency !== NonStandardFrequency.None
        ? val && FrequencyToMultiplier[frequency] * val
        : val;

    const getMaxValue = (
      val?: number,
      frequency?: Frequency | NonStandardFrequency.None
    ) =>
      name && frequency && frequency !== NonStandardFrequency.None
        ? val && Math.round(val / Number(frequency))
        : val;

    switch (paramName) {
      case "min":
        return formatNumber(getMinValue(min, frequency));
      case "max":
        return formatNumber(getMaxValue(max, frequency));
    }
    return undefined;
  };
