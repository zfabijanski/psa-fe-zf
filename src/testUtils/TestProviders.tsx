import { render, RenderOptions } from "@testing-library/react";
import { FC, PropsWithChildren, ReactElement } from "react";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import { Normalize } from "styled-normalize";
import GlobalStyles from "theme/globalStyles";
import { theme } from "theme/theme";
import rootStore, { configureStore, rootReducer as reducer } from "AppStore";
import { BrowserRouter } from "react-router-dom";
import { Modal } from "components/UI/Modal";
import { IntlProvider } from "react-intl";
import { plMessages } from "translations/pl";
import I18nProvider from "I18nProvider";

interface TestProvidersProps {
  store?: typeof rootStore;
  renderModal?: boolean;
  realIntlProvider?: boolean;
}

export const addTestProviders =
  ({
    store = configureStore({ reducer }),
    renderModal = true,
    // @TODO: update tests to use realIntlProvider everywhere
    realIntlProvider = false,
  }: TestProvidersProps): FC<PropsWithChildren<{}>> =>
  ({ children }) => {
    const TranslationsProvider = realIntlProvider ? IntlProvider : I18nProvider;

    return (
      <Provider store={store}>
        <BrowserRouter>
          <ThemeProvider theme={theme}>
            <TranslationsProvider locale="pl" messages={plMessages}>
              <Normalize />
              <GlobalStyles />
              {children}
              {renderModal && <Modal />}
            </TranslationsProvider>
          </ThemeProvider>
        </BrowserRouter>
      </Provider>
    );
  };

const customRender = (
  ui: ReactElement,
  options: Omit<RenderOptions, "options"> = {},
  testProviderOptions: TestProvidersProps = {}
) => render(ui, { wrapper: addTestProviders(testProviderOptions), ...options });

export * from "@testing-library/react";

// override render method
export { customRender as render };
