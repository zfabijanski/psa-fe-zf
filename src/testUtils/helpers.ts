export const sleep = (ms: number = 100): Promise<unknown> =>
  new Promise((resolve) => setTimeout(resolve, ms));
