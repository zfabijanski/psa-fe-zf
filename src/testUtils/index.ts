export * from "./TestProviders";
export * from "./mockCreators";
export * from "./helpers";
