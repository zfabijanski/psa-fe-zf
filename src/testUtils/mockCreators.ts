import {
  ICalculatorCoversConfig,
  IProductCovers,
} from "features/Calculator/types";
import {
  DocumentState,
  IDirectory,
  IDirectoryMetadata,
  IDocumentMetadata,
} from "features/Library/types";
import { IMeeting } from "features/Meeting/History/types";
import { IIllustration, ProductGuid } from "features/Products/types";
import { IModalData, ModalTypes } from "models/Modal";
import {
  Frequency,
  IFundRiskProfileConfig,
  IProductConfig,
  IProductFund,
  IProductGuarantee,
  InvestmentProfile,
  Role,
} from "models/common";
import {
  CalculationStatus,
  Direction,
  ICalculationDetailsDto,
  ICover,
  IInsured,
  YesNo,
} from "models/calculator";

export const createCalculationDetailsDto = (
  overrides?: Partial<ICalculationDetailsDto>
): ICalculationDetailsDto => ({
  calculation_id: 1,
  working_date: "2020-01-01",
  premiums_total: 100,
  guaranteed_death_benefit: 100,
  guaranteed_maturity_benefit: 100,
  projected_maturity_value_egr_m: 100,
  product_categories: [],
  uuid: "uuid",
  version: "version",
  source_system: "source_system",
  product_guid: 1,
  insureds_list: [],
  covers_list: [],
  funds_list: [],
  ...overrides,
});

export const createInsured = (overrides?: Partial<IInsured>): IInsured => ({
  insured_idx: 1,
  insured_uuid: "insured_uuid",
  age_in_months: 1,
  date_of_birth: "2020-01-01",
  role: Role.MAIN,
  ...overrides,
});

export const createCover = (overrides?: Partial<ICover>): ICover => ({
  cover_id: 1,
  cover_idx: 1,
  insured_idx: 1,
  code: "code",
  frequency_code: Frequency.Annually,
  payments: 1,
  direction_of_calculation: Direction.SumAssuredToPremium,
  investment_profile: InvestmentProfile.Balanced,
  indexation: YesNo.Yes,
  sum_assured: 1,
  premium_per_freq: 1,
  duration: 1,
  idx_option: YesNo.Yes,
  rop_option: YesNo.Yes,
  add_sum_assured: 1,
  ending_age: 1,
  exp_mat_ben: 1,
  ...overrides,
});

export const createCalculatorCoversConfig = (
  overrides?: Partial<ICalculatorCoversConfig>
): ICalculatorCoversConfig => ({
  coverName: "coverName",
  isChild: false,
  isHospital: false,
  ...overrides,
});

export const createCalculatorCoversConfigMap = (
  overrides?: Partial<Record<string, ICalculatorCoversConfig>>
): Record<string, ICalculatorCoversConfig> => ({
  coverName1: createCalculatorCoversConfig({ coverName: "coverName1" }),
  coverName2: createCalculatorCoversConfig({ coverName: "coverName2" }),
  ...overrides,
});

export const createProductCovers = (
  overrides?: Partial<IProductCovers>
): IProductCovers => ({
  mainCover: "mainCover",
  additionalCovers: ["additionalCover1", "additionalCover2"],
  wopCovers: ["wopCover1", "wopCover2"],
  childCovers: ["childCover1", "childCover2"],
  ...overrides,
});

export const createProductCoversMap = (
  overrides?: Partial<Record<string, IProductCovers>>
): Record<string, IProductCovers> => ({
  mainCover1: createProductCovers({ mainCover: "mainCover1" }),
  mainCover2: createProductCovers({ mainCover: "mainCover2" }),
  ...overrides,
});

export const createProductConfig = (
  overrides?: Partial<IProductConfig>
): IProductConfig => ({
  product_guid: 1,
  product_name: "product_name",
  product_categories: [],
  product_freqs: [],
  product_covers: [],
  product_funds: [],
  ...overrides,
});

export const createProductConfigMap = (
  overrides?: Partial<Record<string, IProductConfig>>
): Record<string, IProductConfig> => ({
  1: createProductConfig({ product_name: "product_name1" }),
  2: createProductConfig({ product_name: "product_name2" }),
  ...overrides,
});

export const createFundRiskProfileConfig = (
  overrides?: Partial<IFundRiskProfileConfig>
): IFundRiskProfileConfig => ({
  code: "code",
  name_pl: "name_pl",
  name_en: "name_en",
  risk_level: 1,
  status_code: "status_code",
  notes: "notes",
  ...overrides,
});

export const createProductGuarantee = (
  overrides?: Partial<IProductGuarantee>
): IProductGuarantee => ({
  guarantee_code: "guarantee_code",
  guarantee_name: "guarantee_name",
  guarantee_min_value: "1",
  guarantee_duration_months: "4",
  guarantee_percentages: [{ guarantee_percent: 10 }],
  ...overrides,
});

export const createProductFund = (
  overrides?: Partial<IProductFund>
): IProductFund => ({
  cover_code: "cover_code",
  fund_code: "fund_code",
  fund_name: "fund_name",
  fund_risk_profile: "code",
  fund_order: 1,
  guarantees: [createProductGuarantee()],
  ...overrides,
});

export const createModalData = (
  overrides: Partial<IModalData> = {}
): IModalData => ({
  modalIllustration: "alert-fill",
  modalTitleTrKey: "modalTitleTrKey",
  modalContentTrKey: "modalContentTrKey",
  modalContent: "modalContent",
  modalButtons: {
    confirm: { textTrKey: "modal.ok" },
  },
  modalType: ModalTypes.default,
  maskClosable: true,
  ...overrides,
});

export const createIllustration = (
  overrides: Partial<IIllustration> = {}
): IIllustration => ({
  idd_id: 1,
  product_guid: ProductGuid.Savings,
  adequacy_id: 1,
  order_id: 1,
  expected_amount: { value: 3 },
  premium_total: { value: 3 },
  duration: { value: 3 },
  premium_main_cover: { value: 3 },
  frequency: { value: 3 },
  sum_of_premiums: { value: 3 },
  sum_assured: { value: 3 },
  investment_profile: { value: 3 },
  pru_calc_calculation_id: 1,
  status: CalculationStatus.Illustrated,
  json_needs_analysis: null,
  ...overrides,
});

export const createDirectory = (
  overrides?: Partial<IDirectory>
): IDirectory => ({
  directoryId: 1,
  directories: [],
  documentsIds: [1, 2, 3],
  ...overrides,
});

export const createDirectoryMetadata = (
  overrides?: Partial<IDirectoryMetadata>
): IDirectoryMetadata => ({
  directoryId: 1,
  name: "name",
  isOpen: false,
  ...overrides,
});

export const createDocumentMetadata = (
  overrides?: Partial<IDocumentMetadata>
): IDocumentMetadata => ({
  documentId: 1,
  guid: "guid",
  name: "name",
  state: DocumentState.NotAttached,
  isStateChanging: false,
  moduleName: "PRU_LIBRARY",
  ...overrides,
});

export const createMeeting = (overrides?: Partial<IMeeting>): IMeeting => ({
  meetingNo: "meetingNo",
  meetingId: 1,
  name: "name",
  age: 18,
  lastAccessDate: new Date("2023-01-01"),
  createDate: new Date("2023-01-01"),
  statusId: 1,
  statusName: "statusName",
  statusDescription: "statusDescription",
  state: "state",
  haveAdequacy: false,
  haveCalculation: false,
  haveIdd: false,
});
