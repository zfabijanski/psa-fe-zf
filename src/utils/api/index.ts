export * from "./Api";
export * from "./ApiError";
export * from "./SystemError";
export * from "./types";
