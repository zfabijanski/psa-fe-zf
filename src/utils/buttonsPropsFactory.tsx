import { IButtonProps } from "components/UI/PruButton/PruButton";
import { FormattedMessage } from "react-intl";

export const getArrowNextConfig = (id?: string): Partial<IButtonProps> => ({
  icon: "arrow-right",
  iconPosition: "right",
  children: <FormattedMessage id={id} />,
});

export const apkButtonConfig: Partial<IButtonProps> = {
  buttonType: "secondary",
  children: <FormattedMessage id="apk.title" />,
};

export const iddButtonConfig: Partial<IButtonProps> = {
  buttonType: "secondary",
  children: <FormattedMessage id="idd.title" />,
};

export const prevButtonConfig: Partial<IButtonProps> = {
  buttonType: "secondary",
  icon: "arrow-left",
  children: <FormattedMessage id="bottomButtonBar.back" />,
};

export const calculateButtonConfig: Partial<IButtonProps> = {
  buttonType: "secondary",
  htmlType: "submit",
  children: <FormattedMessage id="bottomButtonBar.calculate" />,
};
