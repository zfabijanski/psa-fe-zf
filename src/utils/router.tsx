import { createBrowserHistory, createMemoryHistory } from "history";
import React from "react";
import styled, { StyledComponent } from "styled-components";

export const browserHistory = createBrowserHistory({
  basename: process.env.NODE_ENV !== "test" ? "/psao" : "/",
});
export const memoryHistory = createMemoryHistory();

export type RoutePath =
  | "/"
  | "/about-prudential"
  | "/business-with-prudential"
  | "/business-with-prudential-slides"
  | "/apk"
  | "/calculator"
  | "/adequacy"
  | "/adequacy-idd"
  | "/bop"
  | "/report-bop"
  | "/idd"
  | "/report-idd"
  | "/create"
  | "/history"
  | "/library"
  | "/meeting"
  | "/mail"
  | "/landing";

export const redirect = (
  to: RoutePath,
  shouldChangeBrowserHistory?: boolean,
  params?: object
) => {
  const routingParams = { fromApp: true, ...params };
  if (shouldChangeBrowserHistory) {
    browserHistory.replace(to, routingParams);
  }
  memoryHistory.push(to, routingParams);
};

export const goBack = () => {
  memoryHistory.goBack();
};

const injectRedirect =
  (to: RoutePath, onClick?: React.MouseEventHandler) =>
  (event: React.MouseEvent) => {
    if (typeof onClick === "function") {
      onClick(event);
    }
    redirect(to);
  };

export type RedirectHoC<P extends { onClick?: React.MouseEventHandler }> = (
  C: React.ComponentType<P>,
  to: RoutePath
) => React.ComponentType<P>;

/**
 * HoC injecting redirect to onClick prop
 *
 * @param C - Component (has to use/pass to children "onClick" prop)
 * @param to - Path to redirect to
 *
 * @example
 * interface SpecificProps {
 *   foo?: string;
 * }
 * interface Props extends SpecificProps {
 *   onClick?: React.MouseEventHandler;
 * }
 * const Button: React.FC<Props> = ({ foo, onClick, children }) => (
 *   <button onClick={onClick}>
 *     {foo}
 *     {children}
 *   </button>
 * );
 * const Home = (withRedirect as RedirectHoC<Props>)(Button, "/");
 * const Foo = (withRedirect as RedirectHoC<React.ComponentProps<typeof Button>>)(Button, "/foo");
 */
export const withRedirect: RedirectHoC<
  object & { onClick?: React.MouseEventHandler }
> =
  (C, to) =>
  ({ onClick, ...props }) => {
    return <C {...props} onClick={injectRedirect(to, onClick)} />;
  };

export type PointerHoC<P extends object> = (
  C: React.ComponentType<P>
) => StyledComponent<React.ComponentType<P>, any, {}, never>;

export const withPointer: PointerHoC<object> = (C) => styled(C)`
  cursor: pointer;
`;

export type LinkHoC<P extends { onClick?: React.MouseEventHandler }> = (
  C: React.ComponentType<P>,
  to: RoutePath
) => StyledComponent<React.ComponentType<P>, any, {}, never>;

/**
 * HoC injecting redirect to onClick prop and adding "cursor: pointer" css
 *
 * @param C - Component (has to use/pass to children "onClick" prop)
 * @param to - Path to redirect to
 *
 * @example
 * interface SpecificProps {
 *   foo?: string;
 * }
 * interface Props extends SpecificProps {
 *   onClick?: React.MouseEventHandler;
 * }
 * const Button: React.FC<Props> = ({ foo, onClick, children }) => (
 *   <button onClick={onClick}>
 *     {foo}
 *     {children}
 *   </button>
 * );
 * const Home = (withLink as LinkHoC<Props>)(Button, "/");
 * const Foo = (withLink as LinkHoC<React.ComponentProps<typeof Button>>)(Button, "/foo");
 */
export const withLink: LinkHoC<object> = (C, to) =>
  withPointer(withRedirect(C, to));
