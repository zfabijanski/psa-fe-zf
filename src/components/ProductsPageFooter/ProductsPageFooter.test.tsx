import {
  IProductPageFooterProps,
  ProductsPageFooter,
} from "./ProductsPageFooter";
import { render, screen, fireEvent, sleep } from "testUtils";
import * as router from "utils/router";

const mockOnPrevButtonClick = jest.fn();
const mockOpenConfirmModal = jest.fn();
const mockOnEAppButtonClick = jest.fn();
const mockOnFooterBlockerClick = jest.fn();

const defaultProps: IProductPageFooterProps = {
  isNeedsAnalysisDone: false,
  prevButtonVisible: false,
  onPrevButtonClick: mockOnPrevButtonClick,
  onOpenConfirmModal: mockOpenConfirmModal,
  haveIdd: false,
  onEAppButtonClick: mockOnEAppButtonClick,
  footerBlockerVisible: false,
  onFooterBlockerClick: mockOnFooterBlockerClick,
};

const setup = (props: Partial<IProductPageFooterProps> = {}) =>
  render(<ProductsPageFooter {...defaultProps} {...props} />);

describe("ProductsPageFooter", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it("should show apk title when isNeedsAnalysisDone is true", () => {
    setup({ isNeedsAnalysisDone: true });
    expect(screen.getByText("apk.title")).toBeInTheDocument();
  });

  it("should show back button when prevButtonVisible is true", () => {
    setup({ prevButtonVisible: true });
    expect(screen.getByText("bottomButtonBar.back")).toBeInTheDocument();
  });

  it("should call onPrevButtonClick when prev button is clicked", () => {
    setup({ prevButtonVisible: true });
    fireEvent.click(screen.getByText("bottomButtonBar.back"));
    expect(mockOnPrevButtonClick).toHaveBeenCalledTimes(1);
  });

  it("should navigate to the correct page when idd button is clicked", () => {
    setup({ haveIdd: false });
    fireEvent.click(screen.getByText("idd.title"));
    expect(window.location.pathname).toBe("/adequacy-idd");
  });

  it("should navigate to the correct page when haveIdd is true", () => {
    setup({ haveIdd: true });
    fireEvent.click(screen.getByText("idd.title"));
    expect(window.location.pathname).toBe("/report-idd");
  });

  it("should call onOpenConfirmModal when library button is clicked", () => {
    setup();
    fireEvent.click(screen.getByText("bottomButtonBar.library"));
    expect(mockOpenConfirmModal).toHaveBeenCalledTimes(1);
  });

  it("should call onEAppButtonClick when eApp button is clicked", () => {
    setup();
    fireEvent.click(screen.getByText("bottomButtonBar.application"));
    expect(mockOnEAppButtonClick).toHaveBeenCalledTimes(1);
  });

  it("should call onFooterBlockerClick when footer blocker is clicked", () => {
    setup({ footerBlockerVisible: true });
    fireEvent.click(screen.getByTestId("footer-blocker"));
    expect(mockOnFooterBlockerClick).toHaveBeenCalledTimes(1);
  });

  it("should hide prev button when prevButtonVisible is false", () => {
    setup({ prevButtonVisible: false });
    expect(screen.queryByText("bottomButtonBar.back")).not.toBeInTheDocument();
  });

  it("should show prev button when prevButtonVisible is undefined", () => {
    setup({ prevButtonVisible: undefined });
    expect(screen.getByText("bottomButtonBar.back")).toBeInTheDocument();
  });

  it("should handle onPrevButtonClick being undefined", () => {
    setup({ onPrevButtonClick: undefined, prevButtonVisible: undefined });
    fireEvent.click(screen.getByText("bottomButtonBar.back"));
    expect(mockOnPrevButtonClick).toHaveBeenCalledTimes(0);
  });

  it("should handle onEAppButtonClick being undefined", () => {
    setup({ onEAppButtonClick: undefined });
    fireEvent.click(screen.getByText("bottomButtonBar.application"));
    expect(mockOnEAppButtonClick).toHaveBeenCalledTimes(0);
  });

  it("should handle haveIdd being undefined", () => {
    setup({ haveIdd: undefined });
    fireEvent.click(screen.getByText("idd.title"));
    expect(window.location.pathname).toBe("/adequacy-idd");
  });

  it("should open the modal when the library button is clicked", () => {
    setup();
    fireEvent.click(screen.getByText("bottomButtonBar.library"));
    expect(screen.getByText("bottomButtonBar.library")).toBeInTheDocument();
  });

  it("should call redirect with the correct url when the library button is clicked", async () => {
    const mockRedirect = jest.fn();
    jest.spyOn(router, "redirect").mockImplementation(mockRedirect);
    setup({ onOpenConfirmModal: undefined });
    fireEvent.click(screen.getByText("bottomButtonBar.library"));
    await sleep();
    expect(mockRedirect).toHaveBeenCalledWith("/library");
  });

  it("calls onEAppButtonClick when the eApp button is clicked", () => {
    setup();
    fireEvent.click(screen.getByText("bottomButtonBar.application"));
    expect(mockOnEAppButtonClick).toHaveBeenCalledTimes(1);
  });
});
