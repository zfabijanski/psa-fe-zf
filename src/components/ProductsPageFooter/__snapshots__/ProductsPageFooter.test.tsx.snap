// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsPageFooter should render correctly 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c12 {
  display: inline-block;
  cursor: inherit;
}

.c12 > svg {
  height: 100%;
  display: block;
}

.c10 {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0px;
  cursor: pointer;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 128px;
  max-width: 400px;
  height: 52px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 16px;
  font-weight: 700;
  line-height: 20px;
  text-align: center;
  color: #181818;
  background-color: #FFFFFF;
  border: 2px solid #E3E3E3;
}

.c10 svg g,
.c10 svg g path,
.c10 svg g circle,
.c10 svg g rect {
  stroke: #181818;
}

.c10 svg g rect {
  fill: #181818;
}

.c10 svg g tspan {
  stroke: transparent;
  fill: #181818;
}

.c10 svg > path {
  stroke: #181818;
}

.c10:hover {
  color: #055A60;
  background-color: #FFFFFF;
  border: 2px solid #055A60;
}

.c10:hover svg g,
.c10:hover svg g path,
.c10:hover svg g circle,
.c10:hover svg g rect {
  stroke: #055A60;
}

.c10:hover svg g rect {
  fill: #055A60;
}

.c10:hover svg g tspan {
  stroke: transparent;
  fill: #055A60;
}

.c10:hover svg > path {
  stroke: #055A60;
}

.c10:disabled {
  cursor: initial;
  color: #999999;
  background-color: #F3F3F3;
  border: 2px solid #E3E3E3;
}

.c10:disabled svg g,
.c10:disabled svg g path,
.c10:disabled svg g circle,
.c10:disabled svg g rect {
  stroke: #999999;
}

.c10:disabled svg g rect {
  fill: #999999;
}

.c10:disabled svg g tspan {
  stroke: transparent;
  fill: #999999;
}

.c10:disabled svg > path {
  stroke: #999999;
}

.c10:active:not(:disabled) {
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
}

.c11 {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0px;
  cursor: pointer;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 128px;
  max-width: 400px;
  height: 52px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 16px;
  font-weight: 700;
  line-height: 20px;
  text-align: center;
  color: #FFFFFF;
  background-color: #055A60;
  border: 2px solid #055A60;
}

.c11 svg g,
.c11 svg g path,
.c11 svg g circle,
.c11 svg g rect {
  stroke: #FFFFFF;
}

.c11 svg g rect {
  fill: #FFFFFF;
}

.c11 svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c11 svg > path {
  stroke: #FFFFFF;
}

.c11:hover {
  color: #FFFFFF;
  background-color: #181818;
  border: 2px solid #181818;
}

.c11:hover svg g,
.c11:hover svg g path,
.c11:hover svg g circle,
.c11:hover svg g rect {
  stroke: #FFFFFF;
}

.c11:hover svg g rect {
  fill: #FFFFFF;
}

.c11:hover svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c11:hover svg > path {
  stroke: #FFFFFF;
}

.c11:disabled {
  cursor: initial;
  color: #FFFFFF;
  background-color: #9BBDBF;
  border: 2px solid #9BBDBF;
}

.c11:disabled svg g,
.c11:disabled svg g path,
.c11:disabled svg g circle,
.c11:disabled svg g rect {
  stroke: #FFFFFF;
}

.c11:disabled svg g rect {
  fill: #FFFFFF;
}

.c11:disabled svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c11:disabled svg > path {
  stroke: #FFFFFF;
}

.c11:active:not(:disabled) {
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
}

.c13 {
  margin-right: -4px;
  margin-left: 12px;
}

.c0 {
  margin: 0 auto;
  width: 100%;
  min-height: 80px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  bottom: 0;
  left: 0;
  width: 100%;
  min-height: 100%;
  margin: 0;
  background-color: #FFFFFF;
  z-index: 50;
}

.c3 {
  width: 100%;
  margin: auto;
  padding: 0 48px;
}

.c4 {
  width: 33.33%;
}

.c5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  white-space: nowrap;
}

.c6 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c7 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c9 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 12px;
}

.c8 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

<div>
  <div
    class="c0"
    data-testid="footer"
  >
    <div
      class="c1"
      data-testid="footer-wrapper"
    >
      <div
        class="c2 c3"
      >
        <div
          class="c4"
        >
          <div
            class="c5 c6"
          />
        </div>
        <div
          class="c4"
        >
          <div
            class="c5 c7"
          />
        </div>
        <div
          class="c4"
        >
          <div
            class="c5 c8"
          >
            <div
              class="c9"
            >
              <button
                class="c10"
                type="button"
              >
                bottomButtonBar.library
              </button>
              <button
                class="c10"
                type="button"
              >
                bottomButtonBar.application
              </button>
              <a
                href="/adequacy-idd"
              >
                <button
                  class="c10"
                  type="button"
                >
                  idd.title
                </button>
              </a>
              <a
                href="/apk"
              >
                <button
                  class="c10"
                  type="button"
                >
                  apk.title
                </button>
              </a>
              <button
                class="c11"
                type="button"
              >
                bottomButtonBar.email
                <div
                  class="c12 c13"
                >
                  <svg>
                    arrow-right.svg
                  </svg>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div />
</div>
`;
